pkg/gofr/http/responder_test.go

```diff
@@ -207,6 +208,123 @@ func TestResponder_TemplateResponse(t *testing.T) {
 	assert.Equal(t, expectedBody, responseBody)
 }
 
+func TestResponder_CustomErrorWithResponse(t *testing.T) {
+	w := httptest.NewRecorder()
+	responder := NewResponder(w, http.MethodGet)
+
+	customErr := &CustomError{
+		Code:    http.StatusNotFound,
+		Message: "resource not found",
+		Title:   "Custom Error",
+	}
+
+	responder.Respond(nil, customErr)
+
+	resp := w.Result()
+	defer resp.Body.Close()
+
+	assert.Equal(t, http.StatusNotFound, resp.StatusCode)
+
+	var body map[string]any
+	err := json.NewDecoder(resp.Body).Decode(&body)
+	require.NoError(t, err)
+
+	errorObj, ok := body["error"].(map[string]any)
+	require.Truef(t, ok, "Expected error object to be of type map[string]any, got %T", body["error"])
+
+	assert.Equal(t, "resource not found", errorObj["message"])
+	assert.EqualValues(t, http.StatusNotFound, errorObj["code"])
+	assert.Equal(t, "Custom Error", errorObj["title"])
```

Comment:
I was annoyed by this code.

I was likely to suggest you to use unserialize the json to a struct to avoid all this by hand.

Something like

```go
var body struct {
Message string `json:message`
Code int `json:code`
Errors []string `json:errors`
}
```

But then I realized it could be way simpler to consider using assert.JSONEq directly with the expected returned json